src/meta/processors/parts/GetPartsAllocProcessor.cpp

```diff
@@ -34,8 +34,60 @@ void GetPartsAllocProcessor::process(const cpp2::GetPartsAllocReq& req) {
   }
   handleErrorCode(nebula::cpp2::ErrorCode::SUCCEEDED);
   resp_.set_parts(std::move(parts));
+  auto terms = getTerm(spaceId);
+  if (!terms.empty()) {
+    resp_.set_terms(std::move(terms));
+  }
   onFinished();
 }
 
+std::unordered_map<PartitionID, TermID> GetPartsAllocProcessor::getTerm(GraphSpaceID spaceId) {
+  std::unordered_map<PartitionID, TermID> ret;
+
+  auto spaceKey = MetaServiceUtils::spaceKey(spaceId);
+  auto spaceVal = doGet(spaceKey);
+  if (!nebula::ok(spaceVal)) {
+    auto rc = nebula::error(spaceVal);
+    LOG(ERROR) << "Get Space SpaceId: " << spaceId
+               << " error: " << apache::thrift::util::enumNameSafe(rc);
+    handleErrorCode(rc);
+    return ret;
+  }
+
+  auto properties = MetaServiceUtils::parseSpace(nebula::value(spaceVal));
+  auto partNum = properties.get_partition_num();
+
+  std::vector<PartitionID> partIdVec;
+  std::vector<std::string> leaderKeys;
+  for (auto partId = 1; partId <= partNum; ++partId) {
+    partIdVec.emplace_back(partId);
+    leaderKeys.emplace_back(MetaServiceUtils::leaderKey(spaceId, partId));
+  }
+
+  std::vector<std::string> vals;
+  auto [code, statusVec] = kvstore_->multiGet(0, 0, std::move(leaderKeys), &vals);
+  if (code != nebula::cpp2::ErrorCode::SUCCEEDED &&
+      code != nebula::cpp2::ErrorCode::E_PARTIAL_RESULT) {
+    LOG(INFO) << "error rc = " << apache::thrift::util::enumNameSafe(code);
+    return ret;
```

Comment:
The `handleErrorCode `is missing here

This function returns `ErrorOr<nebula::cpp2::ErrorCode, std::unordered_map<PartitionID, TermID>>` is more appropriate.
In this way, handleErrorCode is handled uniformly outside

Reply:
for now, I consider term is an optional resp for GetPart. So not get a term will not set the GetPart RPC failed.

Reply:
ok